src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1314,14 +1315,21 @@ private function registerAssetMapperConfiguration(array $config, ContainerBuilde
             ->getDefinition('asset_mapper.importmap.manager')
             ->replaceArgument(2, $config['importmap_path'])
             ->replaceArgument(3, $config['vendor_dir'])
-            ->replaceArgument(4, $config['provider'])
+        ;
+
+        $container
+            ->getDefinition('asset_mapper.importmap.resolver')
+            ->replaceArgument(0, $config['provider'])
```

Comment:
not processing the value at compile-time is important to make this configurable by env var